### PR TITLE
[PoC] Co-owners

### PR DESCRIPTION
**Pull request by hipio**
As stated in #166 I would like to have co-owner support in threads, so I'm hereby sending a very crude proof of concept. 

If it's useful to the team, please use it.

### CODE REVIEW

File: modules/ept-frontend/client/admin/moderation/logs.controller.js [hunk 1]
```diff
@@ -174,6 +174,9 @@ var ctrl = ['$rootScope', '$scope', '$location', '$timeout', '$anchorScroll', 'A
     { group: 'Thread Actions',
       value: 'threads.lockPoll',
       desc: 'Locked Thread Poll' },
+    { group: 'Thread Actions',
+      value: 'threads.editCoOwners',
+      desc: 'Edit Owners Thread' },
```
Comment (hipio) [inline modules/ept-frontend/client/admin/moderation/logs.controller.js]: Should be `Edit Co-Owners Thread`

File: modules/ept-threads/migrations/20170201215933-add-multiple-owners.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+var dbm = global.dbm || require('db-migrate');
+var fs = require('fs');
+var path = require('path');
+
+exports.up = function(db, callback) {
+  var filePath = path.join(__dirname + '/sqls/20170201215933-add-co-users-up.sql');
```
Comment (akinsey) [inline modules/ept-threads/migrations/20170201215933-add-multiple-owners.js]: This refers to the sql file as add-co-users-up.sql when the filename is actually add-co-owners-up.sql

Same issue with down file below.

We currently dont support putting migrations in the modules themselves and they are typically written directly into core-pg. We hope to fix that in the future, but for now this shouldn't interfere with us testing

File: modules/ept-frontend/client/components/username_exists/username_exists.directive.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+var directive = ['$timeout', '$q', '$http', function ($timeout, $q, $http) {
```
Comment (akinsey) [inline modules/ept-frontend/client/components/username_exists/username_exists.directive.js]: I wonder if we could leverage a directive that we've already created for this such as the autocomplete-user-id or autocomplete-username.

I know this is a rough proof of concept, but I also noticed the username casing has to be exact with this validator.

See:
https://github.com/epochtalk/epochtalk/tree/master/modules/ept-frontend/client/components/autocomplete_user_id
or
https://github.com/epochtalk/epochtalk/tree/master/modules/ept-frontend/client/components/autocomplete_username

File: modules/ept-bbcode-parser/client/parser.js [hunk 4]
```diff
@@ -1,3 +1,3 @@
 var bbcodeCompiler = require('./bbcode');
 var parser = require('./parseFactory')(bbcodeCompiler);
-module.exports = parser;
+module.exports = parser;
```
Comment (akinsey) [inline modules/ept-bbcode-parser/client/parser.js]: EOF line :p

File: modules/ept-frontend/client/admin/moderation/logs.controller.js [hunk 5]
```diff
@@ -174,6 +174,9 @@ var ctrl = ['$rootScope', '$scope', '$location', '$timeout', '$anchorScroll', 'A
     { group: 'Thread Actions',
       value: 'threads.lockPoll',
       desc: 'Locked Thread Poll' },
+    { group: 'Thread Actions',
```
Comment (akinsey) [inline modules/ept-frontend/client/admin/moderation/logs.controller.js]: There is actually another file that needs to be edited for the action to be actually logged in the moderation log see `/server/plugins/moderation_log/templates.js`. Also the enums for the moderation_action_type would have to be updated, see the latest migration `epochtalk-core-pg/migrations/sqls/20160421232530-user-notes-up.sql`

The mod log should only be updated if a moderator comes in and manually changes the threads owners, see users.update for example `epochtalk/modules/ept-users/routes/update.js` If a mod comes in and edits a user's account then it is logged to the mod log.

Sorry that this isn't well documented yet. I believe we will need permissions for the co-owners routes as well

**Comment by wangbus** [comment]
@hipio Do you have an update for this? If not, I'd like to close this for the time being. (Just some cleaning up here) 👀

**Comment by hipio** [comment]
@wangbus I do not have an update for this. Haven't touched Epochtalk in a while. Feel free to close this PR if you do not want this feature.

**Comment by wangbus** [comment]
@hipio Thanks you for your response. Closing this issue for now. Let's revisit for sure though!

**Pull request closed without merging**